programs/gpl_core/src/instructions/badge.rs

```diff
@@ -19,7 +19,7 @@ pub struct CreateBadge<'info> {
     )]
     pub badge: Account<'info, Badge>,
     #[account(
-        seeds = [Issuer::SEED_PREFIX.as_bytes(), issuer.key().as_ref()],
+        seeds = [Issuer::SEED_PREFIX.as_bytes(), authority.key().as_ref()],
```

Comment:
Who is the authority here? 

Badges are issued by an issuer and the authority of the badge is derived form the authority of the issuer.